pallets/author-inherent/src/lib.rs

```diff
@@ -195,21 +195,40 @@ impl<T: Config> ProvideInherent for Module<T> {
 		Some(Call::set_author(author))
 	}
 
-	fn check_inherent(_call: &Self::Call, data: &InherentData) -> Result<(), Self::Error> {
-		let author_raw = data
-			.get_data::<InherentType>(&INHERENT_IDENTIFIER)
-			.expect("Gets and decodes authorship inherent data")
-			.ok_or_else(|| {
-				InherentError::Other(sp_runtime::RuntimeString::Borrowed(
-					"Decode authorship inherent data failed",
-				))
-			})?;
-		let author =
-			T::AccountId::decode(&mut &author_raw[..]).expect("Decodes author raw inherent data");
-		ensure!(
-			T::CanAuthor::can_author(&author),
-			InherentError::Other(sp_runtime::RuntimeString::Borrowed("Cannot Be Author"))
-		);
+	fn check_inherent(call: &Self::Call, data: &InherentData) -> Result<(), Self::Error> {
+		use sp_std::if_std;
+		if_std!{
+			println!("{:?}", call);
+		}
+		// This if let should always be true. This is the only call that the inehrent could make.
+		if let Self::Call::set_author(claimed_author) = call {
```

Comment:
I don't understand why it is needed to set the author in the check.

Reply:
I'm pattern matching `call` the parameter that was passed into the function against the variants of the `Self::Call` enum.  `set_author` is not actually a function and I'm not calling it. It is a variant of the `Self::Call` enum. This is hard to read because typically enum variants are written in `ThisCaseStandard`. but the `Call` enum is an exception because of how the frame macros work.

Reply:
That might be worthy of a comment, although that's probably true of a lot of other cases...